src/commands.ts

```diff
@@ -513,50 +513,71 @@ export const run = async (functionName: string, cmd: { nondev: boolean }) => {
 
 /**
  * Deploys an Apps Script project.
- * @param version {string} The project version to deploy at.
- * @param description {string} The deployment's description.
+ * @param cmd.versionNumber {string} The project version to deploy at.
+ * @param cmd.desc {string} The deployment description.
+ * @param cmd.deploymentId  {string} The deployment ID to redeploy.
  */
-export const deploy = async (version: number, description = '') => {
+export const deploy = async (cmd :{
+  versionNumber: number,
+  desc: string,
+  deploymentId: string,
+}) => {
   await checkIfOnline();
   await loadAPICredentials();
   const { scriptId } = await getProjectSettings();
   if (!scriptId) return;
   spinner.setSpinnerTitle(LOG.DEPLOYMENT_START(scriptId)).start();
-  async function createDeployment(versionNumber: number) {
-    spinner.setSpinnerTitle(LOG.DEPLOYMENT_CREATE);
-    const deployments = await script.projects.deployments.create({
+
+  let versionNumber = cmd.versionNumber;
+  const description = cmd.desc || '';
+  const deploymentId = cmd.deploymentId;
+
+  // if no version, create a new version
+  if (!versionNumber ){
```

Comment:
Nit: space

`if (!versionNumber) {`

Can you run the linter and see if there are any error?

Reply:
I didn't get an error when running `npm run lint` at my local environment.
Need to add settings?